tools/applicationinsights-web-snippet/src/snippet.ts

```diff
@@ -202,11 +203,52 @@ declare var cfg:ISnippetConfig;
             "js2.cdn.applicationinsights.io",
             "js2.cdn.monitor.azure.com",
             "az416426.vo.msecnd.net" // this domain is supported but not recommended
-        ]
-    
-        // Assigning these to local variables allows them to be minified to save space:
-        let targetSrc : string = (aiConfig as any)["url"] || cfg.src
-        if (targetSrc) {
+        ];
+
+        let targetSrc : string = (aiConfig as any)["url"] || cfg.src;
+        var match = targetSrc.match(/^(http[s]?:\/\/.*\/ai\.)(\d+(\.\d+){0,2})\.(([\w]+\.){0,2}js)$/);
+        console.log("mathc", match);
+        if (match.length === 6 && cfg.sri) {
+            var integrityUrl = match[1] + match[2] + ".integrity.json";
+            var targetType = "@" + match[4];
+            var sender = window.fetch;
+            var integrity: string = null;
+            if (sender && !cfg.useXhr) {
+                // retrieve integrity file using fetch
+                sender(integrityUrl, { method: strGetMethod, mode: "cors" })
+                    .then(response => response.json())
+                    .then(json => {
+                        integrity = json.ext[targetType].integrity;
+                        targetSrc = match[1] + json.ext[targetType].file.substring(3);
+                        setScript(targetSrc, integrity);
+                    })
+                    .catch(error => {
+                        console.error("Error loading JSON:", error);
+                        setScript(targetSrc, integrity); // Fallback to original behavior
+                    });
+            } else if (XMLHttpRequest) {
+                var xhr = new XMLHttpRequest();
+                xhr.open(strGetMethod, integrityUrl);
+                xhr.onreadystatechange = function () {
+                    if (xhr.readyState === XMLHttpRequest.DONE) {
+                        if (xhr.status === 200) {
+                            var json = JSON.parse(xhr.responseText);
+                            integrity = json.ext[targetType].integrity;
+                            targetSrc = match[1] + json.ext[targetType].file.substring(3);
```

Comment:
do we have any better solution instead of calling substring? I am thinking about using [2:] also. Not quite sure which one is better.

Reply:
Lets stick with substring for now as it is supported on all browsers, so it should be supported on the runtimes.

Reply:
Actually, why use substring at all, we should be able to get the regex to give us the domain portion and then this is just a domain + file -- this will also support case (in the future)  where the prefix might not be "ai.3" for example there is already the basic version which is aib.x.min.js (for AISKULight), so we can't guarantee that it will always be that way. And as this is the SDK Loader that gets embedded into applications we won't have the option to "update" it if/when we change it.